routing/per_route_options.go

```diff
@@ -0,0 +1,129 @@
+package routing
+
+import (
+	"fmt"
+	"path/filepath"
+	"regexp"
+	"slices"
+	"sync"
+	"time"
+
+	"github.com/cloudfoundry/cf-test-helpers/v2/cf"
+	"github.com/cloudfoundry/cf-test-helpers/v2/helpers"
+
+	. "github.com/cloudfoundry/cf-acceptance-tests/cats_suite_helpers"
+	"github.com/cloudfoundry/cf-acceptance-tests/helpers/app_helpers"
+	"github.com/cloudfoundry/cf-acceptance-tests/helpers/assets"
+	"github.com/cloudfoundry/cf-acceptance-tests/helpers/random_name"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gexec"
+)
+
+var _ = RoutingDescribe("Per-Route Options", func() {
+	var (
+		appName        string
+		appId          string
+		instanceIds    [2]string
+		leastConnHost  string
+		roundRobinHost string
+	)
+
+	Context("when an app sets the loadbalancing algorithm", func() {
+		BeforeEach(func() {
+			appName = random_name.CATSRandomName("APP")
+			asset := assets.NewAssets()
+			leastConnHost = random_name.CATSRandomName("dora-lc")
+			roundRobinHost = random_name.CATSRandomName("dora-rr")
+			Expect(cf.Cf("push",
+				appName,
+				"-b", Config.GetRubyBuildpackName(),
+				"-m", DEFAULT_MEMORY_LIMIT,
+				"-p", asset.Dora,
+				"--var", fmt.Sprintf("domain=%s", Config.GetAppsDomain()),
+				"--var", fmt.Sprintf("name=%s", appName),
+				"--var", fmt.Sprintf("leastconnhost=%s", leastConnHost),
+				"--var", fmt.Sprintf("roundrobinhost=%s", roundRobinHost),
+				"-f", filepath.Join(asset.Dora, "man_route_opt_lb.yml"),
+			).Wait(Config.CfPushTimeoutDuration())).To(Exit(0))
+			appId = app_helpers.GetAppGuid(appName)
+			appInstanceRegex, _ := regexp.Compile("[[:alnum:]]{8}(-[[:alnum:]]{4}){4}")
+			for i := range 2 {
+				for {
```

Comment:
Can you please change this infinite for loop to be an `Eventually` block instead. That comes with two benefits:
- Times out eventually, if it's not succeeding
- Obviates the need for a manual `time.Sleep`